### PR TITLE
Improve ParametersParser

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php [hunk 1]
```diff
@@ -25,165 +24,113 @@
...
-                    'method' => 'createByParameter',
-                    'arguments' => [
-                        'demo_object',
-                    ],
-                ],
-            ]
-        );
-
-        $expression
-            ->evaluate('service("demo_service")->getWith("bar")', ['container' => $container])
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: Ideal example of "don't mock what you don't own" principle usefulness: `->` does not exist in expression language, `.` should be used instead :)

File: src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php [hunk 2]
```diff
@@ -25,165 +24,119 @@
...
-    function let(ContainerInterface $container, ExpressionLanguage $expression)
+    function it_implements_parameters_parser_interface()
     {
-        $this->beConstructedWith($container, $expression);
+        $this->beConstructedWith(new Container(), new ExpressionLanguage());
+        $this->shouldImplement(ParametersParserInterface::class);
     }
 
-    function it_implements_parameters_parser_interface()
+    function it_should_parse_string_parameters()
```
Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: We don't use `it_should_...` convention for a very long time, why use it here in spite of simple `it_parses_...`?

Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: Just copied the existing ones, you're right.

File: src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php [hunk 3]
```diff
@@ -25,165 +24,119 @@
  */
 final class ParametersParserSpec extends ObjectBehavior
 {
-    function let(ContainerInterface $container, ExpressionLanguage $expression)
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: Why did you remove this method? 
```php
function let()
{
        $this->beConstructedWith(new Container(), new ExpressionLanguage());
}
```

And call `$this->beConstructedWith($container, $expression);` when is needed

Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: 👍 , done!

File: src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php [hunk 4]
```diff
@@ -25,165 +24,104 @@
...
+            ->shouldReturn(['nested' => ['string' => 'Lorem ipsum']])
+        ;
+    }
+
+    function it_parses_boolean_parameters()
+    {
+        $request = new Request();
+        $request->request->set('boolean', true);
+
+        $this->beConstructedWith(new Container(), new ExpressionLanguage());
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: Can be removed

File: src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php [hunk 5]
```diff
@@ -25,165 +24,104 @@
...
+            ->shouldReturn(['nested' => ['boolean' => true]])
+        ;
+    }
+
+    function it_parses_array_parameters()
+    {
+        $request = new Request();
+        $request->request->set('array', ['foo' => 'bar']);
+
+        $this->beConstructedWith(new Container(), new ExpressionLanguage());
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php]: Can be removed 🐋

**Comment by michalmarcinkowski** [comment]
:+1:
